<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Uses phylogenetic ancestral state reconstruction to estimate trait values
for unobserved taxa."><title>Phylogenetic estimation of traits for unobserved taxa — phyEstimate • picante</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Phylogenetic estimation of traits for unobserved taxa — phyEstimate"><meta property="og:description" content="Uses phylogenetic ancestral state reconstruction to estimate trait values
for unobserved taxa."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">picante</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.8.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/skembel/picante/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Phylogenetic estimation of traits for unobserved taxa</h1>
      <small class="dont-index">Source: <a href="https://github.com/skembel/picante/blob/HEAD/R/phyEstimate.R" class="external-link"><code>R/phyEstimate.R</code></a></small>
      <div class="d-none name"><code>phyEstimate.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Uses phylogenetic ancestral state reconstruction to estimate trait values
for unobserved taxa.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">phyEstimate</span><span class="op">(</span><span class="va">phy</span>, <span class="va">trait</span>, method <span class="op">=</span> <span class="st">"pic"</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">phyEstimateDisc</span><span class="op">(</span><span class="va">phy</span>, <span class="va">trait</span>, best.state <span class="op">=</span> <span class="cn">TRUE</span>, cutoff <span class="op">=</span> <span class="fl">0.5</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>phy</dt>
<dd><p>phylo object</p></dd>


<dt>trait</dt>
<dd><p>vector or data.frame containing trait values</p></dd>


<dt>method</dt>
<dd><p>ancestral state estimation method used by <code>ace</code>
(default="pic")</p></dd>


<dt>...</dt>
<dd><p>Additional arguments passed to <code>ace</code></p></dd>


<dt>best.state</dt>
<dd><p>estimate best-supported trait state for discrete
variables? (default=TRUE)</p></dd>


<dt>cutoff</dt>
<dd><p>support cutoff required to declare a best.state</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>phyEstimate produces a data frame with columns:</p>
<dl><dt>est</dt>
<dd><p>Estimated trait value</p></dd>
 <dt>se</dt>
<dd><p>Standard error of estimated trait value</p></dd>
</dl><p>phyEstimateDisc produces a data frame with columns:</p>
<dl><dt>states 1..N</dt>
<dd><p>A
column with statistical support is produced for each discrete trait state</p></dd>

<dt>estimated.state</dt>
<dd><p>If best.state=TRUE, a column with the state with
the highest support</p></dd>
 <dt>estimated.state.support</dt>
<dd><p>Statistical support
for the state with the highest support</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>These functions use phylogenetic ancestral state estimation to infer trait
values for novel taxa on a phylogenetic tree, for continuous
(<code>phyEstimate</code>) and discrete (<code>phyEstimateDisc</code>) traits.</p>
<p>The required input is a phylogenetic tree object plus a vector or
data.frame containing estimated trait values for a subset of the taxa in
the phylogenetic tree. Trait values for taxa that are present in the tree
but not the trait data will be estimated using ancestral state estimation
(Garland and Ives 2000). Briefly, for each taxon present in the tree but
not the trait data, the phylogeny is rerooted at the most recent common
ancestor of the novel taxon and the rest of the phylogeny, and the trait
value of the novel taxon is estimated from the reconstructed trait value at
the root of the rerooted phylogeny.</p>
<p>For <code>phyEstimateDisc</code>, the state with the highest support will be
reported if argument <code>best.state=TRUE</code>. If the best-supported state's
support is less than the specified <code>cutoff</code>, no best state is reported
and a <code>NA</code> value will be returned.</p>
    </div>
    <div class="section level2">
    <h2 id="functions">Functions<a class="anchor" aria-label="anchor" href="#functions"></a></h2>
    
<ul><li><p><code>phyEstimateDisc()</code>: for discrete traits</p></li>
</ul></div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>T. Garland Jr., and A.R. Ives. 2000. Using the past to predict
the present: confidence intervals for regression equations in phylogenetic
comparative methods. American Naturalist 155:346364.</p>
<p>S.W. Kembel, M. Wu, J.A. Eisen, and J.L. Green. 2012. Incorporating 16S
gene copy number information improves estimates of microbial diversity and
abundance. PLoS Computational Biology 8(10):e1002743.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Steven Kembel <a href="mailto:steve.kembel@gmail.com">steve.kembel@gmail.com</a></p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#generate random phylogeny</span></span></span>
<span class="r-in"><span><span class="va">randtree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/rtree.html" class="external-link">rcoal</a></span><span class="op">(</span><span class="fl">50</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co">#simulate trait evolution for a subset of taxa on phylogeny</span></span></span>
<span class="r-in"><span><span class="va">randtraits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/rTraitCont.html" class="external-link">rTraitCont</a></span><span class="op">(</span><span class="va">randtree</span>, sigma<span class="op">=</span><span class="fl">10</span>, root.value<span class="op">=</span><span class="fl">100</span><span class="op">)</span>, <span class="fl">40</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co">#estimate trait values for "missing" taxa using PIC method</span></span></span>
<span class="r-in"><span><span class="fu">phyEstimate</span><span class="op">(</span><span class="va">randtree</span>, <span class="va">randtraits</span>, method<span class="op">=</span><span class="st">"pic"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     estimate        se</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> t38 131.8447 0.6073700</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> t20 132.0236 0.4914396</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> t11 127.8166 0.3214912</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> t23 134.7941 0.2530055</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> t28 135.2075 0.2472558</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> t50 126.5577 1.5057537</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> t18 116.5730 0.3079907</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> t47 117.6162 0.5837055</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> t46 117.9936 0.2125700</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> t39 105.3371 0.4488160</span>
<span class="r-in"><span><span class="va">randtree</span> <span class="op">&lt;-</span> <span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/rtree.html" class="external-link">rcoal</a></span><span class="op">(</span><span class="fl">100</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">db</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>Category <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fl">0</span><span class="op">:</span><span class="fl">1</span>, size <span class="op">=</span> <span class="fl">89</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>label <span class="op">=</span> <span class="va">randtree</span><span class="op">$</span><span class="va">tip.label</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">89</span><span class="op">]</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">data</span><span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span><span class="va">db</span><span class="op">$</span><span class="va">Category</span>, <span class="va">db</span><span class="op">$</span><span class="va">label</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu">phyEstimateDisc</span><span class="op">(</span>trait <span class="op">=</span> <span class="va">data</span>, phy<span class="op">=</span><span class="va">randtree</span>, best.state <span class="op">=</span> <span class="cn">TRUE</span>, cutoff <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] "t31" "t15" "t94" "t11" "t29" "t35" "t49" "t27" "t45" "t42" "t93"</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Steven W. Kembel.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

