<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Phylogenetic estimation of traits for unobserved taxa — phyEstimate • picante</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Phylogenetic estimation of traits for unobserved taxa — phyEstimate"><meta property="og:description" content="Uses phylogenetic ancestral state reconstruction to estimate trait values
for unobserved taxa."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">picante</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.8.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"></ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Phylogenetic estimation of traits for unobserved taxa</h1>
    
    <div class="hidden name"><code>phyEstimate.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Uses phylogenetic ancestral state reconstruction to estimate trait values
for unobserved taxa.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">phyEstimate</span><span class="op">(</span><span class="va">phy</span>, <span class="va">trait</span>, method <span class="op">=</span> <span class="st">"pic"</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>phy</dt>
<dd><p>phylo object</p></dd>


<dt>trait</dt>
<dd><p>vector or data.frame containing trait values</p></dd>


<dt>method</dt>
<dd><p>ancestral state estimation method used by <code>ace</code>
(default="pic")</p></dd>


<dt>...</dt>
<dd><p>Additional arguments passed to <code>ace</code></p></dd>


<dt>best.state</dt>
<dd><p>estimate best-supported trait state for discrete
variables? (default=TRUE)</p></dd>


<dt>cutoff</dt>
<dd><p>support cutoff required to declare a best.state</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>phyEstimate produces a data frame with columns:</p>
<dl><dt>est</dt>
<dd><p>Estimated trait value</p></dd>
 <dt>se</dt>
<dd><p>Standard error of estimated trait value</p></dd>
</dl><p>phyEstimateDisc produces a data frame with columns:</p>
<dl><dt>states 1..N</dt>
<dd><p>A
column with statistical support is produced for each discrete trait state</p></dd>

<dt>estimated.state</dt>
<dd><p>If best.state=TRUE, a column with the state with
the highest support</p></dd>
 <dt>estimated.state.support</dt>
<dd><p>Statistical support
for the state with the highest support</p></dd>

</dl></div>
    <div id="details">
    <h2>Details</h2>
    <p>These functions use phylogenetic ancestral state estimation to infer trait
values for novel taxa on a phylogenetic tree, for continuous
(<code>phyEstimate</code>) and discrete (<code>phyEstimateDisc</code>) traits.</p>
<p>The required input is a phylogenetic tree object plus a vector or
data.frame containing estimated trait values for a subset of the taxa in
the phylogenetic tree. Trait values for taxa that are present in the tree
but not the trait data will be estimated using ancestral state estimation
(Garland and Ives 2000). Briefly, for each taxon present in the tree but
not the trait data, the phylogeny is rerooted at the most recent common
ancestor of the novel taxon and the rest of the phylogeny, and the trait
value of the novel taxon is estimated from the reconstructed trait value at
the root of the rerooted phylogeny.</p>
<p>For <code>phyEstimateDisc</code>, the state with the highest support will be
reported if argument <code>best.state=TRUE</code>. If the best-supported state's
support is less than the specified <code>cutoff</code>, no best state is reported
and a <code>NA</code> value will be returned.</p>
    </div>
    <div id="references">
    <h2>References</h2>
    <p>T. Garland Jr., and A.R. Ives. 2000. Using the past to predict
the present: confidence intervals for regression equations in phylogenetic
comparative methods. American Naturalist 155:346364.</p>
<p>S.W. Kembel, M. Wu, J.A. Eisen, and J.L. Green. 2012. Incorporating 16S
gene copy number information improves estimates of microbial diversity and
abundance. PLoS Computational Biology 8(10):e1002743.</p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Steven Kembel <a href="mailto:steve.kembel@gmail.com">steve.kembel@gmail.com</a></p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#generate random phylogeny</span></span></span>
<span class="r-in"><span><span class="va">randtree</span> <span class="op">&lt;-</span> <span class="fu">rcoal</span><span class="op">(</span><span class="fl">50</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co">#simulate trait evolution for a subset of taxa on phylogeny</span></span></span>
<span class="r-in"><span><span class="va">randtraits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu">rTraitCont</span><span class="op">(</span><span class="va">randtree</span>, sigma<span class="op">=</span><span class="fl">10</span>, root.value<span class="op">=</span><span class="fl">100</span><span class="op">)</span>, <span class="fl">40</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co">#estimate trait values for "missing" taxa using PIC method</span></span></span>
<span class="r-in"><span><span class="fu">phyEstimate</span><span class="op">(</span><span class="va">randtree</span>, <span class="va">randtraits</span>, method<span class="op">=</span><span class="st">"pic"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      estimate        se</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> t1   84.43191 0.5888042</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> t39  75.10473 0.5609304</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> t8   82.00808 0.7994472</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> t31  83.23216 1.0634936</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> t19  83.61038 0.5268943</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> t40  82.70133 0.9560607</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> t9  117.88561 0.5989381</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> t41 100.73610 0.6385890</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> t18  93.99782 0.4286518</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> t27  93.89466 0.4665344</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Steven W. Kembel.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

